stl/inc/source_location

```diff
@@ -0,0 +1,64 @@
+// source_location standard header (core)
+
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#pragma once
+#ifndef _SOURCE_LOCATION_
+#define _SOURCE_LOCATION_
+#include <yvals_core.h>
+#if _STL_COMPILER_PREPROCESSOR
+#if !_HAS_CXX20 || !(defined(__cpp_consteval) || defined(__clang__)) // TRANSITION, LLVM-45745

```

Comment:
We haven't changed this for hours now!

A Clang dev responded to LLVM-4575 to say that they intentionally do not define `__cpp_consteval` because the support for immediate functions in Clang is incomplete and not ready for production use. We should, therefore, not use it:
```suggestion
#if !_HAS_CXX20 || !defined(__cpp_consteval)
```

Reply:
Yes, I think that makes this header effectively dead code at the moment. I suppose we should either bring back the blocked tag, or remove the `defined(__cpp_consteval)` guard and annotate `source_location::current` with `_CONSTEVAL` which allows it to be `constexpr` when the compiler doesn't do `consteval`. Since I suspect the latter would break this feature, I'll add the tag.

Reply:
> We haven't changed this source line for hours now!

Sincerest apologies. I've made 3 commits to make up for that. I also thought to just use `__cpp_consteval` as that I wouldn't have thought that to be defined unconditionally. If that's an incorrect assumption we get to modify that line again! 🎉

Reply:
> I also thought to just use `__cpp_consteval` as I wouldn't have thought that to be defined unconditionally

I left the `_HAS_CXX20` there mostly out of paranoia that some compiler would accidentally define the feature-test macro in a sub-C++20 language mode - not likely, but it has happened before. It's probably not worth the trouble to put it back: we can change this line later if we need to. I think we've established that we can change it over and over whenever we like ;)

Reply:
Interesting... 

Well the main reason I opted for dropping `_HAS_CXX20` was to not have a super long `#else` comment or think of a shorter way to phrase it. 
Seeing as it's going to become just `_HAS_CXX20` eventually (I assume) it seemed like a correct solution that took the minimalist amount of effort.

Reply:
I think the current state (guarding with only `__cpp_consteval`) is fine. For comparison, `<concepts>` is guarded with only `__cpp_lib_concepts`. (I suppose we could change the guard to `__cpp_lib_source_location` to truly centralize everything, but that would be a minor cleanup at best.)